engine/src/main/java/org/terasology/rendering/dag/nodes/DirectionalLightsNode.java

```diff
@@ -152,7 +153,7 @@ private void applyLightBufferPass() {
         bindDisplay();     // TODO: verify this is necessary
         setViewportToSizeOf(sceneOpaque);    // TODO: verify this is necessary
 
-        frameBuffersManager.swapSceneOpaqueFBOs();
+        dynamicFBM.swap(DefaultDynamicFBOs.WriteOnlyGBuffer.getResourceUrn(), DefaultDynamicFBOs.ReadOnlyGBuffer.getResourceUrn());
```

Comment:
Hmm... I have two concerns. 

1. these two FBOs are meant to be swapped, but others are not. So, potentially a node could maliciously or by mistake swap two FBOs that shouldn't be swapped. This might call for "paired" FBOs. Non-paired FBOs can't be swapped.
2. I seem to notice in the swap method that no notification is sent out.  But some nodes need to refresh their sceneOpaque and potentially sceneOpaquePingPong variables.

By the way we'll probably want to move away everywhere from sceneOpaque/sceneOpaquePingPong and use the FBO names instead, that is write/readOnlyGBuffer.